### PR TITLE
Test with different GeoCoders

### PR DESCRIPTION
**Pull request by nigelhorne**
Test the workaround for Github issue #12 
Document the geocoder option
Fix typo that z-axis should have read z-index
Only load G:C:G when we're going to use it
Improved CI testing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,35 +1,21 @@
...
-    - cover
-
-after_success:
-    - cover -report coveralls
-
-sudo: false
-
-notifications:
-    email: false
+  perl Makefile.PL && make && cover -test -report coveralls
```
Comment (leejo) [inline .travis.yml]: What's the reason for changing all of this?

File: lib/HTML/GoogleMaps/V3.pm [hunk 2]
```diff
@@ -71,23 +73,27 @@ Valid options are:
...
 
 sub new {
     my ( $class,%opts ) = @_;
 
-    return bless( {
-        %opts,
-        points     => [],
-        poly_lines => [],
-        geocoder   => Geo::Coder::Google->new(
+    unless($opts{geocoder}) {
```
Comment (leejo) [inline lib/HTML/GoogleMaps/V3.pm]: `if ( ! $opts{geocoder} ) { ...` as unless is a horrible construct other than in single postfix conditions :)

File: lib/HTML/GoogleMaps/V3.pm [hunk 3]
```diff
@@ -71,23 +73,27 @@ Valid options are:
...
+	require Geo::Coder::Google;
+	Geo::Coder::Google->import();
```
Comment (leejo) [inline lib/HTML/GoogleMaps/V3.pm]: just keep the use statement above, no need to do this

**Comment by nigelhorne** [comment]
That's the simple way I prefer to work.  You can ignore it, I know you 
work in a different way.

-Nigel

**Comment by nigelhorne** [comment]
I disagree.  If you keep the 'use' above, the code is loaded even if 
it's never going to be used.

-Nigel

**Comment by nigelhorne** [comment]
That's a religious war I'm not going to enter in to :-)

**Comment by leejo** [comment]
> That's the simple way I prefer to work.  You can ignore it, I know you
work in a different way.

The repo has a functioning .travis.yml file already so that's why i was confused by the change.

**Comment by leejo** [comment]
> I disagree.  If you keep the 'use' above, the code is loaded even if it's never going to be used.

It's going to be used by 99.99% of the use of this module.

**Comment by leejo** [comment]
> That's a religious war I'm not going to enter in to :-)

Not really, in correct English grammar we don't use "unless" when we mean "if" and that's why my brain fails to parse it as a leading conditional. Not only that, the danger of using `unless` as a leading conditional is someone coming along later and introducing a second condition, which can eventually lead to a mess like:

```perl
unless ( ! $foo || $bar ) { ...
```

Do not use `unless` as a leading conditional, it stops you having to fix it later.

**Comment by nigelhorne** [comment]
Also, it was a good test to ensure that the geocode option was actually 
being used to override the default one - it wasn't until I added that code.

**Pull request closed without merging**